commit/plugin_e2e_test.go

```diff
@@ -0,0 +1,194 @@
+package commit
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/smartcontractkit/chainlink-ccip/chainconfig"
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/testhelpers"
+	"github.com/smartcontractkit/chainlink-ccip/internal/mocks"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	reader_mock "github.com/smartcontractkit/chainlink-ccip/mocks/internal_/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+)
+
+func TestPlugin_E2E(t *testing.T) {
+	destChain := ccipocr3.ChainSelector(123)
+	sourceChain1 := ccipocr3.ChainSelector(1)
+	sourceChain2 := ccipocr3.ChainSelector(2)
+
+	ctx := context.Background()
+	lggr := logger.Test(t)
+
+	nodeIDs := []commontypes.OracleID{1, 2, 3}
+	peerIDs := []libocrtypes.PeerID{{1}, {2}, {3}}
+
+	oracleIDToP2pID := make(map[commontypes.OracleID]libocrtypes.PeerID)
```

Comment:
nit: maybe we can move this and `chainConfig` definitions to a `setup` func that returns all of this data in a struct?

Reply:
Made many changes to the test since u reviewed, lmk if you still have the same suggestion